pkg/install.bzl

```diff
@@ -0,0 +1,167 @@
+# Copyright 2021 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""rules for creating install scripts from pkg_filegroups and friends.
+
+This module provides an interface (`pkg_install`) for creating a `bazel
+run`-able installation script.
+
+"""
+
+load("//:providers.bzl", "PackageDirsInfo", "PackageFilegroupInfo", "PackageFilesInfo", "PackageSymlinkInfo")
+load("//private:pkg_files.bzl", "process_src", "write_manifest")
+load("@rules_python//python:defs.bzl", "py_binary")
+
+def _pkg_install_script_impl(ctx):
+    script_file = ctx.actions.declare_file(ctx.attr.name + ".py")
+
+    fragments = []
+    files_to_run = []
+    content_map = {}
+    for src in ctx.attr.srcs:
+        if DefaultInfo in src:
+            files_to_run.append(src[DefaultInfo].files)
+
+        process_src(content_map, src, src.label, "0644", None, None)
+
+    manifest_file = ctx.actions.declare_file(ctx.attr.name + "-install-manifest.json")
+
+    # Write out the manifest in terms of "short" paths, which are those expected
+    # when you make `bazel run`nable binaries).
+    #
+    # Note that these paths are different when used as tools run within a build.
+    # See also
+    # https://docs.bazel.build/versions/4.1.0/skylark/rules.html#tools-with-runfiles
+    write_manifest(ctx, manifest_file, content_map, use_short_path = True)
+
+    # Get the label of the actual py_binary used to run this script.
+    #
+    # This is super brittle, but I don't know how to otherwise get this
+    # information without creating a circular dependency given the current state
+    # of rules_python.
+    binary_label_str = "{}//{}:{}".format(
+        ctx.label.workspace_name,
+        ctx.label.package,
+        # The name of the binary is the name of this target, minus
+        # "_install_script".
+        ctx.label.name[:-len("_install_script")],
+    )
+
+    ctx.actions.expand_template(
+        template = ctx.file._script_template,
+        output = script_file,
+        substitutions = {
+            "##MANIFEST_INCLUSION##": manifest_file.short_path,
```

Comment:
I'm chuckling to myself. You use ##var##. I like ${VAR} (or sometimes @VAR@).
We also have  "%DIRECTORY_ROOT% in the filter_directory tests and plain {VAR} in the toolchains.

I don't want to hold you up while we debate what the best style is, but let's not introduce a new one just for this feature.

Reply:
Hah, sure.  I was trying to ensure that elements of the template couldn't be interpreted as code in editors when working on the script.  Turns out it wasn't necessary.

I'll go with `{VAR}` for now (it's what was in the old `pkg_rpm` rule), and we can converge on a single option later.  I'm partial to `%VAR%` or `@VAR@` myself.